tests/Rule/UrlTest.php

```diff
@@ -44,6 +44,36 @@ public function testReturnsFalseOnInvalidUrls($value, $error)
         $this->assertEquals($expected, $this->validator->getMessages());
     }
 
+    public function testFailsOnNotWhiteListedScheme()
+    {
+        $this->validator->required('url')->url(['http', 'https']);
+
+        $this->assertFalse(
+            $this->validator->validate([
+                'url' => 'git://github.com'
```

Comment:
I would add ``mailto:`` as a testcase as well, don't know how good ``parse_url`` is :)